source/Halibut/Transport/SecureListener.cs

```diff
@@ -38,7 +38,8 @@ public class SecureListener : IDisposable
         readonly ILogFactory logFactory;
         readonly Func<string> getFriendlyHtmlPageContent;
         readonly Func<Dictionary<string, string>> getFriendlyHtmlPageHeaders;
-        readonly CancellationTokenSource cts = new();
+        readonly CancellationTokenSource cts;
+        readonly CancellationToken cancellationToken;
```

Comment:
Usage of a single cancellation token to avoid creating new cancellation tokens each time they are needed from the source (it's a simple pass-through object but we create a lot of them)

Reply:
Oh so the CTS creates a new token each time it is asked for a token?

Reply:
Yes, it's a very lightweight wrapper that calls into the CancellationTokenSource, but it's another object with another reference.